XiEditor/StatusBar.swift

```diff
@@ -0,0 +1,241 @@
+// Copyright 2018 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import Foundation
+import Cocoa
+
+enum StatusItemAlignment: String {
+    case left = "left"
+    case right = "right"
+}
+
+class StatusItem: NSTextField {
+    let key: String
+    var value: String = ""
+    let barAlignment: StatusItemAlignment
+
+    init(_ key: String, _ value: String, _ barAlignment: String) {
+        self.key = key
+        self.value = value
+        self.barAlignment = StatusItemAlignment(rawValue: barAlignment)!
+        super.init(frame: .zero)
+
+        // Similar to what NSTextField's label convenience init creates
+        self.isEditable = false
+        self.isSelectable = false
+        self.textColor = NSColor.labelColor
+        self.backgroundColor = NSColor.clear
+        self.lineBreakMode = .byClipping
+        self.isBezeled = false
+        self.stringValue = value
+    }
+
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+}
+
+class StatusBar: NSView {
+
+    var currentKeys = [String]()
+    var hiddenItems = [StatusItem]()
+    var leftItems = [StatusItem]()
+    var rightItems = [StatusItem]()
+    var currentItems: [StatusItem] {
```

Comment:
instead of storing current keys and current items separately, can we just have a dictionary of keys: items?